### PR TITLE
Equivalence of operators

### LINKED ISSUE #29524: equivalence of x &&= y and x && ( x = y )
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND_assignment

### What specific section or headline is this issue about?

Description

### What information was incorrect, unhelpful, or incomplete?

Worth mentioning that x &&= y is equivalent to x && ( x = y ) except that x is only evaluated once.

### What did you expect to see?

Elaboration that x is evaluated only once in x &&= y.

### Do you have any supporting links, references, or citations?

Caveat that in x &&= y, x is evaluated only once, contrary to x && ( x = y ).

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/operators/logical_and_assignment`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND_assignment
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/operators/logical_and_assignment/index.md
* Last commit: https://github.com/mdn/content/commit/f616cb604af851f77f8cd59368e94ee3e43a8838
* Document last modified: 2023-08-15T14:59:44.000Z

</details>

### PR DESCRIPTION
**Pull request by manviii27**
### Description

Modified the docs to mention the following:
1. x &&= y is equivalent to x && ( x = y ) except that x is only evaluated once.
2. x ||= y is equivalent to x || ( x = y ) except that x is only evaluated once.
3. that x ??= y is equivalent to x ?? ( x = y ) except that x is only evaluated once.

### Motivation

 To contribute to mdn docs and these changes resolve the issue mentioned below which helps readers understand JS better.

### Additional details

Issue #29524

### Related issues and pull requests

 Fixes #29524

### CODE REVIEW

**Comment by Josh-Cena** [comment]
Closing in favor of https://github.com/mdn/content/pull/29563

**Comment by Josh-Cena** [comment]
@manviii27 Your PR was slightly later than #29563, which is why I closed yours. However, neither this one nor that one is complete (I asked in #29524 for _all_ compound assignments to be updated), and I have to say that one is in a worse condition. In case we don't see an immediate improvement from that one, I am very glad to reopen. In the meantime you can still push to this branch.

**Comment by manviii27** [comment]
@Josh-Cena I'll make the required changes. I'll be glad to help if the other PR fails to merge

**Comment by Josh-Cena** [comment]
If you really want to, I'm open to change everything to inline (i.e. `x &&= y` is equivalent to `x && (x = y)`). The only reason it's done this way is because the logical compounds are harder to parse while things like `x = x + y` are super trivial, but changing everything to block seems like overkill.

**Comment by manviii27** [comment]
Makes sense. I will see to it and make the changes

**Comment by Josh-Cena** [comment]
Hi @manviii27 we're going to work on your PR instead. Thanks for opening it!

**Comment by manviii27** [comment]
@Josh-Cena I've made the changes, please review and let me know. :)

**Comment by Josh-Cena** [comment]
Sorry, a bit occupied at the moment. I'll definitely merge this before the end of week.

File: files/en-us/web/javascript/reference/operators/logical_and_assignment/index.md [hunk 1]
```diff
@@ -19,12 +19,7 @@ x &&= y
...
 ## Description
 
-Logical AND assignment [_short-circuits_](/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence#short-circuiting), meaning that `x &&= y` is equivalent to:
-
-```js
-x && (x = y);
-```
-
+Logical AND assignment [_short-circuits_](/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence#short-circuiting), meaning that `x &&= y` is equivalent to `x && (x = y)`, except that the expression `x` is only evaluated once.
 No assignment is performed if the left-hand side is not truthy, due to short-circuiting of the [logical AND](/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND) operator. For example, the following does not throw an error, despite `x` being `const`:
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/operators/logical_and_assignment/index.md:23]:
```suggestion

No assignment is performed if the left-hand side is not truthy, due to short-circuiting of the [logical AND](/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND) operator. For example, the following does not throw an error, despite `x` being `const`:
```

File: files/en-us/web/javascript/reference/operators/logical_or_assignment/index.md [hunk 2]
```diff
@@ -19,12 +19,7 @@ x ||= y
...
 ## Description
 
-Logical OR assignment [_short-circuits_](/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence#short-circuiting), meaning that `x ||= y` is equivalent to:
-
-```js
-x || (x = y);
-```
-
+Logical OR assignment [_short-circuits_](/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence#short-circuiting), meaning that `x ||= y` is equivalent to `x || (x = y)`, except that the expression `x` is only evaluated once.
 No assignment is performed if the left-hand side is not falsy, due to short-circuiting of the [logical OR](/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR) operator. For example, the following does not throw an error, despite `x` being `const`:
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/operators/logical_or_assignment/index.md:23]:
```suggestion

No assignment is performed if the left-hand side is not falsy, due to short-circuiting of the [logical OR](/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR) operator. For example, the following does not throw an error, despite `x` being `const`:
```

File: files/en-us/web/javascript/reference/operators/nullish_coalescing_assignment/index.md [hunk 3]
```diff
@@ -19,12 +19,7 @@ x ??= y
...
 ## Description
 
-Nullish coalescing assignment [_short-circuits_](/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence#short-circuiting), meaning that `x ??= y` is equivalent to:
-
-```js
-x ?? (x = y);
-```
-
+Nullish coalescing assignment [_short-circuits_](/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence#short-circuiting), meaning that `x ??= y` is equivalent to `x ?? (x = y)`, except that the expression `x` is only evaluated once.
 No assignment is performed if the left-hand side is not nullish, due to short-circuiting of the [nullish coalescing](/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) operator. For example, the following does not throw an error, despite `x` being `const`:
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/operators/nullish_coalescing_assignment/index.md:23]:
```suggestion

No assignment is performed if the left-hand side is not nullish, due to short-circuiting of the [nullish coalescing](/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) operator. For example, the following does not throw an error, despite `x` being `const`:
```

**Review by Josh-Cena — APPROVED** [review]
Thanks!

**Comment by Josh-Cena** [comment]
Congrats on your first contribution—welcome aboard!

**Comment by manviii27** [comment]
Thank you!

**Pull request merged**